[10.0][ADD] Cronjob to auto update Gravatar

hmrodrigues opened the pull request:
It adds a new field to the user to enable/disable gravatar autoupdate.
This feature can be disabled if the ir.config_parameter gravatar.autoupdate is set to False

Portuguese translations were also updated on base_user_gravatar

Signed-off-by: Hugo Rodrigues <hugorodrigues@openmailbox.org>

Commit:
[ADD] Cronjob to auto update Gravatar

It adds a new field to the user to configure enable/disable gravatar
autoupdate.
This feature can be disabled if the ir.config_parameter
gravatar.autoupdate is set to False

Portuguese translations were also updated

Signed-off-by: Hugo Rodrigues <hugorodrigues@openmailbox.org>

elicoidal commented on README.md:
```diff
@@ -38,7 +38,7 @@ addon | version | summary
 [base_search_fuzzy](base_search_fuzzy/) | 10.0.1.0.0 | Fuzzy search with the PostgreSQL trigram extension
```
No need to update this file: it is auto updated every night

elicoidal commented on base_user_gravatar/models/res_users.py:
```diff
@@ -39,3 +58,18 @@ def get_gravatar_image(self):
+        IrParameter = self.env['ir.config_parameter']
+        auto_update = IrParameter.search([('key', '=', 'gravatar.autoupdate')], limit=1)
+        if auto_update and auto_update.value in ('True', 'true', '1'):
+            _logger.info('Starting gravatar update')
```
Gravatar

hmrodrigues replied:
Do I a create a second commit or squash into just one?

elicoidal reviewed (changes requested):
some details. other than LGTM

Garamotte commented on base_user_gravatar/i18n/base_user_gravatar.pot:
```diff
@@ -0,0 +1,50 @@
+# Translation of Odoo Server.
```
Remove the pot file.

hmrodrigues replied:
It's supposed to leave the old pot file?

Garamotte replied:
No, all pot files are removed from modules in OCA repositories.

Garamotte commented on base_user_gravatar/models/res_users.py:
```diff
@@ -1,19 +1,38 @@
+    @api.multi
+    def _compute_gravatar_autoupdate_enabled(self):
+        IrParameter = self.env['ir.config_parameter']
+        auto_update = IrParameter.search([('key', '=', 'gravatar.autoupdate')], limit=1)
```
Use `IrParameter.get_param('gravatar.autoupdate')`

Garamotte commented on base_user_gravatar/models/res_users.py:
```diff
@@ -39,3 +58,18 @@ def get_gravatar_image(self):
+    @api.model
+    def _update_gravatars(self):
+        IrParameter = self.env['ir.config_parameter']
+        auto_update = IrParameter.search([('key', '=', 'gravatar.autoupdate')], limit=1)
```
Use `IrParameter.get_param('gravatar.autoupdate')`

Commit:
[FIX] PR sugested changes

Revert README.md
Replaced gravatar by Gravatar on log
Removed pot file
Using IrParameter.get_param('gravatar.autoupdate') instead of search

Signed-off-by: Hugo Rodrigues <hmr@thinkopensolutions.pt>

hmrodrigues commented:
I just commit the changes. I don't know if I need to squash into a single commit or if it's ok to have two commits. If I need to squash, please tell me so I can do it.

elicoidal commented:
@hmrodrigues Better squash the commits whenever possible in order to reduce the footprint

Please check travis too

lasley commented:
Please don't squash commits into already reviewed commits though- reviewers will typically look at the diffs in order to skip having to recover the full code again. 

Once everything is said and done, the merger can squash on merge, or you can squash at the logical change sets - whichever you prefer.

hmrodrigues commented:
Regarding pylint, I don't know what to the executable files, are they supposed to be executable? And the error regarding the import error, what should I use?
The line size error, I thought that Odoo ignored PEP8 E501

Sorry about the questions, first time contributing to OCA.

moylop260 commented:
```bash
base_user_gravatar/tests/__init__.py:1: [W8201(incoherent-interpreter-exec-perm), ] You have a python file with execution permissions but you don't have a interpreter magic comment. If you really needs a execution permission then add a magic comment ( https://en.wikipedia.org/wiki/Shebang_(Unix) ). If you don't needs a execution permission then remove it with: chmod -x __init__.py
base_user_gravatar/tests/test_res_users.py:1: [W8201(incoherent-interpreter-exec-perm), ] You have a python file with execution permissions but you don't have a interpreter magic comment. If you really needs a execution permission then add a magic comment ( https://en.wikipedia.org/wiki/Shebang_(Unix) ). If you don't needs a execution permission then remove it with: chmod -x test_res_users.py
```

Run the command `chmod -x base_user_gravatar/tests/test_res_users.py base_user_gravatar/tests/__init__.py`

hmrodrigues commented:
@moylop260 ok so it doesn't need executable permissions, and about the import error? I don't want to mess up with the tests

moylop260 commented:
```patch
- from odoo.addons.base_user_gravatar.models.res_users import ResUsers
+ from ..models.res_users import ResUsers
```

You can find more OCA guidelines [here](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#python)

hmrodrigues commented:
@moylop260 oh thanks

Commit:
[FIX] Fixed flake8 and pylint

Signed-off-by: Hugo Rodrigues <hmr@thinkopensolutions.pt>

Commit:
[FIX] Fixed more flake8

Signed-off-by: Hugo Rodrigues <hmr@thinkopensolutions.pt>

hmrodrigues commented:
Hey all,
How do I fix the codecov tests? I never used codecov so I don't know how it works.

lasley commented:
Hi @hmrodrigues - CodeCov isn't testing, but is instead giving an output of how much our test coverage is dropping with the merge of a given PR. I use the [browser plugin](https://docs.codecov.io/docs/browser-extension), which makes it easy to identify hot spots. I'll comment on the misses here inline

lasley commented on base_user_gravatar/models/res_users.py:
```diff
@@ -1,19 +1,39 @@
+
+    @api.multi
+    def _compute_gravatar_autoupdate_enabled(self):
+        IrParameter = self.env['ir.config_parameter']
```
This method is untested

lasley commented on base_user_gravatar/models/res_users.py:
```diff
@@ -39,3 +59,19 @@ def get_gravatar_image(self):
+
+    @api.model
+    def _update_gravatars(self):
+        IrParameter = self.env['ir.config_parameter']
```
This method is untested

Commit:
[TEST] _update_gravatars

Signed-off-by: Hugo Rodrigues <hmr@thinkopensolutions.pt>

hmrodrigues committed:
[TEST] cron and compute

hmrodrigues committed:
Merge branch '10.0' into 10.0

hmrodrigues committed:
[FIX] wrong model

hmrodrigues committed:
[REM] Removed option to disable auto update

If the user changes the gravatar, it makes sense to change on Odoo
